TeamCode/src/main/java/org/firstinspires/ftc/teamcode/modules/robot/DriveToAprilTag.java

```diff
@@ -92,145 +86,93 @@
-
-                // Determine heading, range and Yaw (tag image rotation) error so we can use them to control the robot automatically.
-                double  rangeError      = (desiredTag.ftcPose.range - DESIRED_DISTANCE);
-                double  headingError    = desiredTag.ftcPose.bearing;
-                double  yawError        = desiredTag.ftcPose.yaw;
-
-                // Use the speed and turn "gains" to calculate how we want the robot to move.
-                drive  = Range.clip(-rangeError * SPEED_GAIN, -MAX_AUTO_SPEED, MAX_AUTO_SPEED);
-                turn   = Range.clip(headingError * TURN_GAIN, -MAX_AUTO_TURN, MAX_AUTO_TURN) ;
-                strafe = Range.clip(yawError * STRAFE_GAIN, -MAX_AUTO_STRAFE, MAX_AUTO_STRAFE);
-
-                telemetry.addData("Auto","Drive %5.2f, Strafe %5.2f, Turn %5.2f ", drive, strafe, turn);
-            } else {
-
-                // drive using manual POV Joystick mode.  Slow things down to make the robot more controlable.
-                drive = Math.pow(-gamepad1.left_stick_y, 3);
-                strafe = Math.pow(gamepad1.left_stick_x, 3);
-                turn = Math.pow(gamepad1.right_stick_x, 3);
-                mecanumDrive.drive(drive, strafe, turn, imu.getHeading(AngleUnit.RADIANS));
-
-                telemetry.addData("Manual","Drive %5.2f, Strafe %5.2f, Turn %5.2f ", drive, strafe, turn);
-            }
-            telemetry.update();
-
-            // Apply desired axes motions to the drivetrain.
-            mecanumDrive.drive(drive, strafe, turn);
-            sleep(10);
+        // Determine heading, range and Yaw (tag image rotation) error so we can use them to control the robot automatically.
+        double rangeError = (desiredTag.ftcPose.range - DESIRED_DISTANCE);
+        double headingError = desiredTag.ftcPose.bearing;
+        double yawError = desiredTag.ftcPose.yaw;
+
+        // Use the speed and turn "gains" to calculate how we want the robot to move.
+        forward = Range.clip(rangeError * SPEED_GAIN, -MAX_AUTO_SPEED, MAX_AUTO_SPEED);
+        rotate = Range.clip(headingError * TURN_GAIN, -MAX_AUTO_TURN, MAX_AUTO_TURN);
+        strafe = Range.clip(-yawError * STRAFE_GAIN, -MAX_AUTO_STRAFE, MAX_AUTO_STRAFE);
+
+        // Apply desired axes motions to the drivetrain.
+        if((
+                Math.abs(forward) > MIN_SPEED || Math.abs(strafe) > MIN_SPEED || Math.abs(rotate) > MIN_SPEED) &&
+                targetFound
+        ){
+            drive.drive((float) forward, (float) -strafe, (float) -rotate);
+            return false;
+        }else{
+            drive.drive(0,0,0);
+            return true;
+        }
+    }
+    public void telemetry() {
+        if(desiredTag != null) {
+            telemetry.addData("found tag", desiredTag.id);
         }
     }
-
     /**
      * Initialize the AprilTag processor.
      */
-    private void initAprilTag() {
+    private void initAprilTag(LinearOpMode op) {
```

Comment:
You should pass in the camera here not the op mode or make the camera part of the state of this class.